scripts/test.sh

```diff
@@ -488,6 +507,19 @@ function bom_pass {
     "${modules[@]}")
   code="$?"
 
+  # Procfs is not used in darwin (osx) builds, thus we need to include it manually in BOM check. See https://github.com/etcd-io/etcd/pull/19599 for more details.
+  if [ "$OS" = "darwin" ] ; then
+    output=$(echo "$output" | jq --tab '. + [{
+      "project": "github.com/prometheus/procfs",
+      "licenses": [
+        {
+          "type": "Apache License 2.0",
+          "confidence": 1
+        }
+      ]
+    }] | sort_by(.project)')
+  fi
```

Comment:
Thanks for the changes, @Dogacel. I don't think this is a clean solution, as we'll also need to implement something similar for the `fix` target. Could you explore an alternative by adding this license to the `bill-of-materials-override.json`, and removing this workaround?

Reply:
Unfortunately I couldn't make `override` work as `procfs` should be excluded in the BOM file rather than being overwritten. There is no way to delete entries using override file [(Source)](https://github.com/appscodelabs/license-bill-of-materials/blob/6018e0c5287c58796f780a14f5ee3b13c4e0d0fe/license-bill-of-materials.go#L634-L643). Also running `fix` just removes `procfs` from the BOM 😢

Reply:
:face_with_head_bandage: I suggest removing the `bom` part from your pull request. If you remove that, I think we should be able to merge it, and there will be less friction in merging as the change is relatively small.

Then, in a follow-up pull request, we can address the `bom` issue. A possible solution would be to skip verifying and updating the `bom` if the architecture is other than amd64 (I wonder if Linux ARM64 would fail, too).

Reply:
Makes sense, I also wonder if we can generate the BOM targeting the amd64 architecture in Go compiler using `GOARCH` params, maybe we can work with the BOM generator library to support that.

As you said, I'll separate that part and open another PR.

Reply:
Reverted the BOM workaround 👍 PTAL 🙂